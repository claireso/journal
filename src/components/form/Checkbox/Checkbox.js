import React from 'react'
import PropTypes from 'prop-types'

import { GroupInline } from '../Group'
import Label from '../Label'

const Checkbox = (props = {}) => {
  return (
    <GroupInline>
      <Label htmlFor={props.name}>{props.label}</Label>
      <input id={props.name} type="checkbox" name={props.name} defaultChecked={props.value} />
    </GroupInline>
  )
}

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.bool
}

export default Checkbox
