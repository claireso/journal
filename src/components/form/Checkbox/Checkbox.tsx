import React from 'react'

import { GroupInline } from '../Group'
import Label from '../Label'

interface CheckboxProps {
  name: string
  label: string
  value: boolean
}

const Checkbox = ({ name, label, value = false }: CheckboxProps) => {
  return (
    <GroupInline>
      <Label htmlFor={name}>{label}</Label>
      <input id={name} type="checkbox" name={name} defaultChecked={value} />
    </GroupInline>
  )
}

export default Checkbox
