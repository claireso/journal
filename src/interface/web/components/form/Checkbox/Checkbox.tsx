import React from 'react'

import Group from '../Group'
import Label from '../Label'

interface CheckboxProps {
  name: string
  label: string
  value?: boolean
}

const Checkbox = ({ name, label, value = false }: CheckboxProps) => {
  return (
    <Group inline>
      <Label htmlFor={name}>{label}</Label>
      <input id={name} type="checkbox" name={name} defaultChecked={value} />
    </Group>
  )
}

export default Checkbox
