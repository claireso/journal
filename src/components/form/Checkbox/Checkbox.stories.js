import React, { Fragment } from 'react'

import Checkbox from './index'

export default {
  title: 'Form/Checkbox'
}

export const Basic = () => (
  <Fragment>
    <Checkbox label="Label" name="portrait" />
    <Checkbox label="Label" name="landscape" value={true} />
  </Fragment>
)
