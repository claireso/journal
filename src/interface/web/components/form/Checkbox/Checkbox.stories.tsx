import React, { Fragment } from 'react'

import type { Meta, StoryObj } from '@storybook/react'

import Checkbox from './Checkbox'

const meta: Meta<typeof Checkbox> = {
  title: 'Form/Checkbox',
  component: Checkbox
}

export default meta
type Story = StoryObj<typeof Checkbox>

export const Basic: Story = {
  render() {
    return (
      <Fragment>
        <Checkbox label="Label" name="portrait" />
        <Checkbox label="Label" name="landscape" value={true} />
      </Fragment>
    )
  }
}
