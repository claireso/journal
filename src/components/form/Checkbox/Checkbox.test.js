import { render } from '@testing-library/react'

import Checkbox from './index'

describe('<Checkbox />', () => {
  const props = {
    label: 'Checkbox',
    value: false,
    name: 'checkbox'
  }

  test('should render checkbox', () => {
    const { container } = render(<Checkbox {...props} />)

    expect(container).toMatchSnapshot()
  })

  test('should render checked checkbox', () => {
    const { container } = render(<Checkbox {...props} value={true} />)

    expect(container).toMatchSnapshot()
  })
})
