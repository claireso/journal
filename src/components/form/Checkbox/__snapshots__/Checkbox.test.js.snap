// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render checkbox 1`] = `
.c0 {
  margin: 0 0 var(--gutter);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > label {
  margin: 0 1rem 0 0;
}

.c2 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

<div>
  <div
    class="c0 c1"
  >
    <label
      class="c2"
      for="checkbox"
    >
      Checkbox
    </label>
    <input
      id="checkbox"
      name="checkbox"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`<Checkbox /> should render checked checkbox 1`] = `
.c0 {
  margin: 0 0 var(--gutter);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > label {
  margin: 0 1rem 0 0;
}

.c2 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

<div>
  <div
    class="c0 c1"
  >
    <label
      class="c2"
      for="checkbox"
    >
      Checkbox
    </label>
    <input
      checked=""
      id="checkbox"
      name="checkbox"
      type="checkbox"
    />
  </div>
</div>
`;
